packages/hooks/src/useRequestPhoneNumberOTP.tsx

```diff
@@ -19,22 +20,28 @@ const useRequestPhoneNumberOTP = () => {
         ...rest
     } = useMutation('phone_number_challenge');
     const [error_message, setErrorMessage] = React.useState<React.ReactNode>('');
+    const { client } = useStore();
+    const { verification_code } = client;
+    const { phone_number_verification } = verification_code;
     const {
         mutation: { mutateAsync: updateSettings },
     } = useSettings();
 
     //TODOs: need to wait confirmation from the team whether to stay at phone number page when refresh or restart the email verification process again
     const requestOnSMS = () => {
         mutate({
-            payload: { carrier: VERIFICATION_SERVICES.SMS, email_code: localStorage.getItem('email_otp_code') || '' },
+            payload: {
+                carrier: VERIFICATION_SERVICES.SMS,
+                email_code: phone_number_verification || localStorage.getItem('email_otp_code') || '',
```

Comment:
why do we need to store email code in localstorage?
since we route user to phone number screen only from email screen,  do we need to store value in localStorage?

Reply:
There might be a scenario where user's route from email link and also users redirect to the `ConfirmYourPhoneNumber` page with manual input of otp. The localstorage is to cover the scenario where user's redirect through manual input of otp.

Reply:
Maybe i can reuse the `setVerificationCode` from client-store here to set the manual otp input users 🤔